File: restful-server/src/main/java/via/sep/restful_server/notification/dto/PriceChangeNotificationDTO.java
package via.sep.restful_server.notification.dto;

import com.fasterxml.jackson.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class PriceChangeNotificationDTO {
    private String propertyId;
    private String address;
    private BigDecimal oldPrice;
    private BigDecimal newPrice;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;
}

File: Blazor/RealEstateBlazor/Data/DTOs/PriceChangeNotificationDTO.cs
using RealEstateBlazor.Services;

namespace RealEstateBlazor.Data.DTOs;

public class PriceChangeNotificationDTO : INotification
{
    public string PropertyId { get; set; }
    public string Address { get; set; }
    public decimal OldPrice { get; set; }
    public decimal NewPrice { get; set; }
    public DateTime Timestamp { get; set; }
}